qiskit/backends/ibmq/ibmqprovider.py

```diff
@@ -149,6 +149,16 @@ def remove_account(self, token, url=QE_URL, **kwargs):
         if not removed:
             raise IBMQAccountError('Unable to find credentials')
 
+    def remove_all_accounts(self, delete=True):
```

Comment:
@ajavadia wanted it to be remove_accounts with the s i dont care either way. Thanks for doing this so fast. 
I give ali a bit to comment otherwise i will approve :-)

Reply:
Yeah can we have it `remove_accounts()`?

I would actually love to later delete `remove_account()`, and just have `remove_accounts()` with filter (similar to #962). It is quite cumbersome that you have to supply `token` & `url` every time you want to add/load/remove stuff (it is akin to using the old Qconfig way, which we're trying to avoid). Instead, you should be able to just say `remove_accounts(hub='myhub')` and all accounts matching that would be deleted.

For this reason, it is preferable if `remove_accounts()` relied on `remove_account()`, so it can easily replace it later. I don't think the speed of execution is a big deal here.

Reply:
While I think the difference between remove_account() and remove_accounts() is too small, and people are going to likely typo and misread it all the time, I understand your point here. I'll push up a new rev in a second.